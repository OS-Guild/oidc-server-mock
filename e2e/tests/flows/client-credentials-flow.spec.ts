import * as dotenv from 'dotenv';

import clients from '../../config/clients-configuration.json';
import type { Client } from '../../types';
import { introspectEndpoint, tokenEndpoint } from '../../helpers';

describe('Client Credentials Flow', () => {
  let client: Client;
  let token: string;

  beforeAll(() => {
    dotenv.config();
    client = clients.find(c => c.ClientId === 'client-credentials-flow-client-id');
    expect(client).toBeDefined();
  });

  test('Token Endpoint', async () => {
    const parameters = {
      client_id: client.ClientId,
      client_secret: client.ClientSecrets?.[0],
      grant_type: 'client_credentials',
      scope: client.AllowedScopes,
    };

    token = await tokenEndpoint(parameters);
  });

  test('Introspection Endpoint', async () => {
    await introspectEndpoint(token, 'some-app');
  });
});
