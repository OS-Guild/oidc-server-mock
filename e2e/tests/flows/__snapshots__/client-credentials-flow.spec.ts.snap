// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client Credentials Flow Introspection Endpoint 1`] = `
{
  "iss": "http://localhost:8080",
  "aud": "some-app",
  "client_id": "client-credentials-flow-client-id",
  "string_claim": "string_claim_value",
  "json_claim": [
    "value1",
    "value2"
  ],
  "active": true,
  "scope": "some-app-scope-1"
}
`;

exports[`Client Credentials Flow Token Endpoint 1`] = `
{
  "alg": "RS256",
  "typ": "JWT",
  "iss": "http://localhost:8080",
  "aud": "some-app",
  "client_id": "client-credentials-flow-client-id",
  "string_claim": "string_claim_value",
  "scope": [
    "some-app-scope-1"
  ],
  "json_claim": [
    "value1",
    "value2"
  ]
}
`;
